clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+   * @throws TachyonException if an unexpected exception is thrown
+   */
+  TachyonURI createDirectory(TachyonURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Creates a directory.
+   *
+   * @param path the path of the directory to create in Tachyon space
+   * @param options options to associate with this operation
+   * @return the {@link TachyonURI} referencing the newly created directory
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  TachyonURI createDirectory(TachyonURI path, CreateDirectoryOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #createFile(TachyonURI, CreateFileOptions)} with default options.
+   *
+   * @param path the path of the file to create in Tachyon space
+   * @return a {@link FileOutStream} which will write data to the newly created file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileOutStream createFile(TachyonURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Creates a file.
+   *
+   * @param path the path of the file to create in Tachyon space
+   * @param options options to associate with this operation
+   * @return a {@link FileOutStream} which will write data to the newly created file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileOutStream createFile(TachyonURI path, CreateFileOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #delete(TachyonURI, DeleteOptions)} with default options.
+   *
+   * @param path the path to delete in Tachyon space
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws DirectoryNotEmptyException if recursive is false and the path is a nonempty directory
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void delete(TachyonURI path)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Deletes a file or a directory.
```

Comment:
For the convenience methods, I feel the same with @apc999 that it would be better to be explicit, otherwise, users need to go to the javadoc of options class, and see what default values it provides.